src/test/scala/beam/router/PhyssymXmlToOsmConverter.scala

```diff
@@ -0,0 +1,269 @@
+    private val timeStampStr = DateTimeFormatter.ISO_INSTANT.format(zoned)
+
+    def writeToFile(path: Path): Unit = {
+      logger.info(s"Writing conversion in the file [$path]")
+      val xmlOutputFactory = XMLOutputFactory.newInstance
+
+      val xmlStreamWriter: XMLStreamWriter = xmlOutputFactory.createXMLStreamWriter(new FileOutputStream(path.toFile))
+      xmlStreamWriter.writeStartDocument(StandardCharsets.UTF_8.name(), "1.0")
+      xmlStreamWriter.writeCharacters(breakLine)
+      xmlStreamWriter.writeStartElement("osm")
+      xmlStreamWriter.writeAttribute("version", "0.6")
+      xmlStreamWriter.writeAttribute("generator", "beam")
+      xmlStreamWriter.writeCharacters(breakLine)
+
+      writeXmlNodes(xmlStreamWriter, nodes)
+      ways.foreach{ way =>
+        xmlStreamWriter.writeCharacters(indentation)
+        xmlStreamWriter.writeStartElement("way")
+
+        xmlStreamWriter.writeAttribute("id", way.id)
+        xmlStreamWriter.writeAttribute("version", "1")
+        xmlStreamWriter.writeAttribute("timestamp", timeStampStr)
+        xmlStreamWriter.writeCharacters(breakLine)
+
+        writeWayTags(xmlStreamWriter, way)
+
+        xmlStreamWriter.writeCharacters(indentation)
+        xmlStreamWriter.writeEndElement()
+        xmlStreamWriter.writeCharacters(breakLine)
+      }
+      xmlStreamWriter.writeEndDocument()
+      logger.info(s"Finished conversion. File: [$path]")
+    }
+
+    private def writeWayTags(xmlStreamWriter: XMLStreamWriter, way: Way): Unit = {
+      writeNd(xmlStreamWriter, way.from)
+      writeNd(xmlStreamWriter, way.to)
+
+      way.attributes.foreach {
+        case (prop, value) =>
+          xmlStreamWriter.writeCharacters(indentation * 2)
+
+          xmlStreamWriter.writeEmptyElement("tag")
+          xmlStreamWriter.writeAttribute("k", prop)
+          xmlStreamWriter.writeAttribute("v", value)
+
+          xmlStreamWriter.writeCharacters(breakLine)
+      }
+    }
+
+    private def writeNd(xmlStreamWriter: XMLStreamWriter, value: String): Unit = {
+      xmlStreamWriter.writeCharacters(indentation * 2)
+
+      xmlStreamWriter.writeEmptyElement("nd")
+      xmlStreamWriter.writeAttribute("ref", value)
+
+      xmlStreamWriter.writeCharacters(breakLine)
+
+    }
+
+    private def writeXmlNodes(xmlStreamWriter: XMLStreamWriter, nodes: Seq[Node]): Unit = {
+      nodes.foreach { node =>
+        xmlStreamWriter.writeCharacters(indentation)
+
+        xmlStreamWriter.writeEmptyElement("node")
+
+        xmlStreamWriter.writeAttribute("id", node.id)
+
+        xmlStreamWriter.writeAttribute("version", "1")
+        xmlStreamWriter.writeAttribute("timestamp", timeStampStr)
+
+        xmlStreamWriter.writeAttribute("lat", node.wgsCoordinate.latitude.toString)
+        xmlStreamWriter.writeAttribute("lon", node.wgsCoordinate.longitude.toString)
+
+        xmlStreamWriter.writeCharacters(breakLine)
+      }
+    }
+  }
+
+  def main(args: Array[String]): Unit = {
```

Comment:
nitpick: an example of run so it is easy to use

Reply:
changes done and waiting for your review